func_sig_registry/utils/abi.py

```diff
@@ -139,3 +139,24 @@ def event_definition_to_text_signature(abi: Dict[str, Any]) -> str:
             for abi_input in abi.get('inputs', [])]
         ),
     )
+
+
+def retrieve_stats_from_import_results(import_results):
+        num_processed = len(import_results)
+        
+        import_results = [
+            result
+            for result in import_results
+            if result is not None
+        ]
+
+        num_ignored = num_processed - len(import_results)
+
+        if len(import_results) == 0:
+            num_imported = 0
+            num_duplicates = 0
+        else:
+            num_imported = sum(tuple(zip(*import_results))[1])
+            num_duplicates = len(import_results) - num_imported
+        
+        return num_processed, num_imported, num_duplicates, num_ignored
```

Comment:
A nice way to do this would be to use a named tuple

https://docs.python.org/3/library/collections.html#collections.namedtuple

It is a sort of self documenting way to have a drop-in replacement for places where you have tuples of values so that you don't have to remember ordering or what each value means.

```
ImportStats = namedtuple('ImportStats', ('num_processed', 'num_imported', 'num_duplicates', 'num_ignored'))

...

return ImportStats(
    num_processed=num_processed,
    num_imported=num_imported,
    ...
)

stats = retrieve_stats_from_import_results(...)
stats.num_processed
stats.num_imported
```

Much nicer to work with in my experience.  We used named tuples A LOT in our code for this exact type of thing.